### PR TITLE
Bug fix #5038: Dependent Module Jars Are Not Added When Debugging

### LINKED ISSUE #5038: Dependent Module Jars Are Not Added When Debugging With IDEA
The temporary directory that is created for debugging does not contain all of the dependencies from the `build.gradle` in the `lib` sub-directory.

In my build file I have:

    dependencies {
      compile 'com.microsoft.azure.functions:azure-functions-java-library:1.4.2'    
      compile project(":model")
      compile "org.apache.camel:camel-core:$camel_version"

That project dependency is not included when debugging but it is included when building with gradle.

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java [hunk 1]
```diff
@@ -229,14 +229,24 @@ public static void copyLocalSettingsToStagingFolder(Path stagingFolder,
...
-        final Path jarFile = JarUtils.buildJarFileToStagingPath(stagingFolder.toString(), module);
+        final Path jarFile;
+        // test if it is gradle project
+        final IntellijGradleFunctionProject gradleProject = new IntellijGradleFunctionProject(project, module);
+        if (gradleProject.isValid() && gradleProject.getArtifactFile() != null) {
+            jarFile = gradleProject.getArtifactFile().toPath();
+            gradleProject.packageJar();
+            FileUtils.copyFileToDirectory(gradleProject.getArtifactFile(), stagingFolder.toFile());
+        } else {
+            jarFile = JarUtils.buildJarFileToStagingPath(stagingFolder.toString(), module);;
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java:247]: remove one comma;

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java [hunk 2]
```diff
@@ -251,21 +261,26 @@ public static void copyLocalSettingsToStagingFolder(Path stagingFolder,
         copyFilesWithDefaultContent(hostJson, hostJsonFile, DEFAULT_HOST_JSON);
 
         final List<File> jarFiles = new ArrayList<>();
-        OrderEnumerator.orderEntries(module).productionOnly().forEachLibrary(lib -> {
-            if (StringUtils.isNotEmpty(lib.getName()) && ArrayUtils.contains(lib.getName().split("\\:"), FUNCTION_JAVA_LIBRARY_ARTIFACT_ID)) {
-                return true;
-            }
+        if (gradleProject.isValid()) {
+            jarFiles.addAll(gradleProject.getDependencies());
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java:265]: remove the copy for FUNCTION_JAVA_LIBRARY_ARTIFACT_ID

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java [hunk 3]
```diff
@@ -229,14 +229,24 @@ public static void copyLocalSettingsToStagingFolder(Path stagingFolder,
...
         }
 
-        final Path jarFile = JarUtils.buildJarFileToStagingPath(stagingFolder.toString(), module);
+        final Path jarFile;
+        // test if it is gradle project
+        final IntellijGradleFunctionProject gradleProject = new IntellijGradleFunctionProject(project, module);
+        if (gradleProject.isValid() && gradleProject.getArtifactFile() != null) {
+            jarFile = gradleProject.getArtifactFile().toPath();
+            gradleProject.packageJar();
+            FileUtils.copyFileToDirectory(gradleProject.getArtifactFile(), stagingFolder.toFile());
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java:245]: check file exists of gradleProject.getArtifactFile()

**Review by wangmingliang-ms — APPROVED** [review]
LGTM

**Pull request merged**